### PR TITLE
Posts

### PR DESCRIPTION
**Pull request by Alex-M1**

### CODE REVIEW

File: .eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...],[396,398],[398,400],[438,440],[440,442],[458,460],[25,27],[61,63],[84,86],[121,123],[162,164],[198,200],[202,204],[18,20],[35,37],[63,65],[89,91],[92,94],[51,53],[84,86],[116,118],[165,167],[211,213],[278,280],[317,319],[351,353],[353,355],[389,391],[476,478],[519,521],[530,532],[587,589],[611,613],[682,684],[691,693],[769,771],[837,839],[863,865],[916,918],[935,937],[993,995],[1000,1002],[1006,1008],[1008,1010],[1088,1090],[1131,1133],[1188,1190],[1206,1208],[1273,1275],[1280,1282],[1282,1284],[1358,1360],[1387,1389],[1456,1458],[1463,1465],[1509,1511],[1538,1540],[1567,1569],[1582,1584],[1611,1613],[1619,1621],[1662,1664],[1668,1670],[1671,1673],[41,43],[81,83],[83,85],[125,127],[137,139],[156,158],[177,179],[196,198],[202,204],[217,219],[236,238],[257,259],[263,265],[276,278],[296,298],[317,319],[323,325],[334,336],[353,355],[371,373],[377,379],[392,394],[411,413],[429,431],[435,437],[445,447],[464,466],[482,484],[488,490],[499,501],[518,520],[536,538],[542,544],[552,554],[571,573],[589,591],[595,597],[606,608],[625,627],[646,648],[668,670],[674,676],[704,706],[706,708],[752,754],[65,67],[102,104],[104,106],[127,129],[217,219],[255,257],[271,273],[280,282],[286,288],[289,291]]
```
Comment (ArtemVakulenko42) [inline .eslintcache:1]: what is this?

File: README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
 ```
 {
   "login":"admin",
-  "password": "admin123"
+  "password": "Admin123"
 }
 ```
 Все поля обязательны. Поле логин можно вводить только латинские символы и цифры <br>
```
Comment (ArtemVakulenko42) [inline README.md:12]: why in russian?

File: src/commonTypes/controllers.ts [hunk 3]
```diff
@@ -1,4 +1,7 @@
-import { Response } from 'express';
+import { Response, Request } from 'express';
+import { ParamsDictionary, Query } from 'express-serve-static-core';
 
 export type TControllerReturn = Response<any, Record<string, any>>;
-export type TRoles = 'User' | 'Admin' | 'SuperAdmin';
+export type TRoles = 'User' | 'Manager' | 'SuperAdmin';
+export type TPostStatus = 'private' | 'pending' | 'public' | 'rejected';
```
Comment (ArtemVakulenko42) [inline src/commonTypes/controllers.ts:6]: why no enum value?

File: src/constants/messages.ts [hunk 4]
```diff
@@ -6,4 +6,5 @@ export const MESSAGES = {
   invalid_login: 'invalid_login',
   abstract_err: 'something_wrong',
   un_autorized: 'un_autorized',
```
Comment (ArtemVakulenko42) [inline src/constants/messages.ts:8]: without space

File: src/constants/posts.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export const PER_PAGE = 10;
+export const INITIAL_PAGE = 1;
+export enum MessageStatus {
```
Comment (ArtemVakulenko42) [inline src/constants/posts.ts:3]: change to MessagesStatusEnum

**Review by ArtemVakulenko42 — COMMENTED** [review]
fix this shit

File: .eslintcache (hunk 1 above)
Comment (SkypnikMR) [reply .eslintcache:1]: Eslintcache, ты шо, не видишь

File: src/controllers/Posts/Posts.controller.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    }
+  };
+
+  private findPosts = async (query, options?: IFindPostOptions) => {
+    const page = +query.page || INITIAL_PAGE;
+    const perPage = +query.per_page || PER_PAGE;
+    const range = page * perPage;
+    let posts = await Post.find(options || {}, { __v: false, status: false })
+      .populate({ path: 'author', select: 'login' });
+    posts = posts.reverse().filter((_post: IPosts, i: number) => i >= range - perPage && i < range);
```
Comment (SkypnikMR) [inline src/controllers/Posts/Posts.controller.ts:76]: мерзость

File: src/helpers/validations.ts [hunk 7]
```diff
@@ -26,6 +26,7 @@ export const tokenValidation = (req: Request, res: Response, next: NextFunction)
   jwt.verify(token, process.env.TOKEN_SECRET, (err: Error, user) => {
     if (err) return res.sendStatus(401).json(MESSAGES.un_autorized);
     req.body.userId = user.userId;
+    req.body.userRole = user.role;
```
Comment (SkypnikMR) [inline src/helpers/validations.ts:29]: like

**Review by SkypnikMR — APPROVED** [review]

**Pull request merged**